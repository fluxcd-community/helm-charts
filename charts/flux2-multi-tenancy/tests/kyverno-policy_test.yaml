suite: test kyverno deployment
templates:
  - kyverno-policy.yaml
tests:
  - it: should have kind Secret for k8s >= 1.19
    capabilities:
      majorVersion: 1
      minorVersion: 19
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: ClusterPolicy
      - isAPIVersion:
          of: kyverno.io/v1
  - it: should match snapshot of default values
    asserts:
      - matchSnapshot: {}
