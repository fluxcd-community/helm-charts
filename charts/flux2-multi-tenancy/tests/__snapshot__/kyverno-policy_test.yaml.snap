should match snapshot of default values:
  1: |
    apiVersion: kyverno.io/v1
    kind: ClusterPolicy
    metadata:
      annotations:
        helm.sh/hook: post-install
      labels:
        app.kubernetes.io/instance: NAMESPACE
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/part-of: flux
        helm.sh/chart: flux2-multi-tenancy-0.0.4
      name: RELEASE-NAME
      namespace: NAMESPACE
    spec:
      rules:
      - exclude:
          resources:
            namespaces:
            - flux-system
        match:
          resources:
            kinds:
            - Kustomization
            - HelmRelease
        name: serviceAccountName
        validate:
          message: .spec.serviceAccountName is required
          pattern:
            spec:
              serviceAccountName: ?*
      - exclude:
          resources:
            namespaces:
            - flux-system
        match:
          resources:
            kinds:
            - Kustomization
        name: kustomizationSourceRefNamespace
        preconditions:
          any:
          - key: '{{request.object.spec.sourceRef.namespace}}'
            operator: NotEquals
            value: ""
        validate:
          deny:
            conditions:
            - key: '{{request.object.spec.sourceRef.namespace}}'
              operator: NotEquals
              value: '{{request.object.metadata.namespace}}'
          message: .spec.sourceRef.namespace must be the same as metadata.namespace
      - exclude:
          resources:
            namespaces:
            - flux-system
        match:
          resources:
            kinds:
            - HelmRelease
        name: helmReleaseSourceRefNamespace
        preconditions:
          any:
          - key: '{{request.object.spec.chart.spec.sourceRef.namespace}}'
            operator: NotEquals
            value: ""
        validate:
          deny:
            conditions:
            - key: '{{request.object.spec.chart.spec.sourceRef.namespace}}'
              operator: NotEquals
              value: '{{request.object.metadata.namespace}}'
          message: .spec.chart.spec.sourceRef.namespace must be the same as metadata.namespace
      validationFailureAction: enforce
